main/src/com/google/refine/browsing/Engine.java

```diff
@@ -91,7 +91,7 @@ public void setMode(Mode mode) {
         _mode = mode;
     }
 
-    public FilteredRows getAllRows() {
+    public synchronized FilteredRows getAllRows() {
```

Comment:
Is this needed in addition to the front end change to show the busy dialog or is it redundant?

While it doesn't make sense to have multiple operations per client, there may be multiple clients, so we don't want this single threaded unless there are real synchronization issues to protect against.